src/main/java/com/cloudbees/jenkins/plugins/bitbucket/api/BitbucketAuthenticator.java

```diff
@@ -0,0 +1,101 @@
+package com.cloudbees.jenkins.plugins.bitbucket.api;
+
+import com.cloudbees.jenkins.plugins.bitbucket.endpoints.BitbucketCloudEndpoint;
+import com.cloudbees.plugins.credentials.common.StandardCredentials;
+import jenkins.authentication.tokens.api.AuthenticationTokenContext;
+import org.apache.http.HttpHost;
+import org.apache.http.HttpRequest;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+/**
+ * Support for various different methods of authenticating with Bitbucket
+ */
+public abstract class BitbucketAuthenticator {
+
+    private String id;
+
+    /**
+     * The URL protocol as reported in an {@link AuthenticationTokenContext}
+     */
+    public static final String PROTOCOL_PURPOSE = "PROTOCOL";
+
+    /**
+     * The purpose value for HTTP instances
+     */
+    public static final String PROTOCOL_HTTP = "HTTP";
```

Comment:
just use the URI scheme rather than a constant (i.e. parse the URL, or just substring up to the first `:` and lowercase the scheme part)... if you want to have constants for HTTP and HTTPS their values should be the same as an URL (i.e. lowercase)